src/VisualStudio/Core/Impl/ProjectSystem/CPS/CPSProject_IWorkspaceProjectContext.cs

```diff
@@ -282,5 +289,8 @@ public void RemoveAnalyzerConfigFile(string filePath)
             => _projectSystemProject.RemoveAnalyzerConfigFile(filePath);
 
         public IAsyncDisposable CreateBatchScope() => _projectSystemProject.CreateBatchScope();
+
+        public async ValueTask<IAsyncDisposable> CreateBatchScopeAsync(CancellationToken cancellationToken)
```

Comment:
>CreateBatchScopeAsync [](http://example.com/codeflow?start=49&length=21)

Could CPS just use this instead of StartBatchAsync and then not need EndBatchAsync?

Reply:
After talking with Drew, the assumption is that CPS can be modified to just use CreateBatchScopeAsync. I'll try to get a frankenbuild on my machine with the changes in CPS to test against this PR and will wait to elevate out of draft status until 

1) I can at least verify the behavior based on the frankenbuild.
2) Both Cyrus/Drew have had a chance to look at the core idea here.